skbio/metadata/tests/test_interval.py

```diff
@@ -446,6 +446,34 @@ def test_init_upper_bound_lt_lower_bound(self):
         with self.assertRaises(ValueError):
             IntervalMetadata(-1)
 
+    def test_upper_bound_is_none(self):
```

Comment:
I thought I had addressed this comment? I can't use `__eq__` to directly assert equality of `IntervalMetadata` in these cases (eg line 471) because `__eq__` also compares `upper_bound`, which is not the same between the expected and observed objects. Thus, I have to use `_intervals`.